src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -607,7 +607,7 @@ bool QgsOgrProvider::setSubsetString( const QString &theSQL, bool updateFeatureC
   invalidateCachedExtent( false );
 
   // Changing the filter may change capabilities
-  computeCapabilities();
+  emit capabilitiesNeedUpdate();
```

Comment:
why to make a signal for that ? Nothing outside of the provider is interested in that. Or does it cause some asynchronous call ?

Reply:
yes, that was the idea: we don't want the update capabilities function to be called on first call of setSubsetString but only for following changes that would occour to the filter. If you prefer, we could add an argument to setSubsetString to tell wether the capabilities needs to be updated or not.
But maybe we could just call computeCapabilities all the times and remove the explicit call to computeCapabilities from the open() code, the effect would be the same.

Reply:
> we could add an argument to setSubsetString to tell wether the capabilities needs to be updated or not.

Would seem more obvious to me. signals are hard to follow

Reply:
mmm, setSubsetString is an override, we would need to change all providers.  I actually added an asynchronous call to avoid recursion when setSubSetString is invoked from open(), the update would call open() again and again for shapefiles.
I think the signal is our only chance in this case (I'd really prefer to not change all providers API for such a particular case).

Reply:
oh well, so be it. some code comments to capture this will help later readers of that code :-)

Reply:
@rouault I spoke too early, see my last commit for a connect-less implementation (easier to backport too).
What do you think?

Reply:
> What do you think?

+1